cachi2/core/package_managers/bundler/parser.py

```diff
@@ -4,8 +4,10 @@
 from functools import cached_property
 from pathlib import Path
 from typing import Annotated, Optional, Union
+from urllib.parse import urlparse
```

Comment:
**Commit message:**

the search may return different results in the future (e.g. they rename the function), so I think you just want to point it to: https://github.com/rubygems/rubygems/blob/0bcc770cc623d2da98b15389cf3f3aeaf5efae6f/bundler/lib/bundler/source/git.rb#L347
OR
https://github.com/rubygems/rubygems/blob/0bcc770cc623d2da98b15389cf3f3aeaf5efae6f/bundler/lib/bundler/plugin/api/source.rb#L112

(both deal with source URL handling and I didn't want to spend time digging into how they're related, so the former should be enough of a reference for our purposes)

Also,
 > The name is constructed from name of the
> git repository (not actual name of the gem)

this should also be backed by a reference, again, you probably want https://github.com/rubygems/rubygems/blob/0bcc770cc623d2da98b15389cf3f3aeaf5efae6f/bundler/lib/bundler/source/git.rb#L339 but one has to first parse the regex they use in their head when reading the commit, so going back again to https://github.com/rubygems/rubygems/blob/0bcc770cc623d2da98b15389cf3f3aeaf5efae6f/bundler/lib/bundler/plugin/api/source.rb#L110 which is aligned with your implementation. I don't care which one you go with as long as we reference both how base name and the ref are computed.